spec/cell_spec.rb

```diff
@@ -0,0 +1,116 @@
+require 'cell'
+
+describe Cell do
+
+  let(:cell){ Cell.new }
+
+  describe '#new' do
+    it 'is initialized without a mine' do
+      expect(cell.mine).to be false
+    end
+    it 'is initialized as uncleared' do
+      expect(cell.cleared).to be false
+    end
+    it 'is initialized without a flag' do
+      expect(cell.flagged).to be false
+    end
+    it 'is initialized with adjacent_mines equal to zero' do
+      expect(cell.adjacent_mines).to be 0
+    end
+  end
+
+  describe '#place_mine' do
+    it 'adds a mine to the cell' do
+      cell.place_mine
+      expect(cell.mine).to be true
+    end
+  end
+
+  describe '#count_adjacent_mine' do
+    it 'adds to the count' do
+      cell.count_adjacent_mine
+      expect(cell.adjacent_mines).to eq 1
+    end
+    it 'accepts multiple neighboring mines' do
+      4.times { cell.count_adjacent_mine }
+      expect(cell.adjacent_mines).to eq 4
+    end
+  end
+
+  describe '#clear' do
+    context 'without a mine' do
+      it 'clears the cell if there is no flag' do
+        cell.clear
+        expect(cell.cleared).to be true
+      end
+      it 'does nothing if the cell has a flag' do
+        cell.flag
+        cell.clear
+        expect(cell.cleared).to be false
```

Comment:
I think I may have had something like this happen in my code but the play for some reason would still go.... I wonder if you had that issue where it wouldn't clear if the player did a play?